src/drivers/net/lan91c111/lan91c111.c

```diff
@@ -164,55 +78,93 @@ static void lan91c111_push_data(uint8_t *data, int len) {
 	}
 }
 
-static int lan91c111_xmit(struct net_device *dev, struct sk_buff *skb) {
+static int lan91c111_send_hw(struct sk_buff *skb) {
 	uint8_t packet_num;
+	uint16_t packet_len;
 	uint8_t *data;
+
+	lan91c111_set_bank(2);
+
+	packet_num = REG8_LOAD(BANK_TX_ALLOC) & PNUM_MASK;
+
+	REG8_STORE(BANK_PNR, packet_num);
+
+	/* Write header */
+	REG16_STORE(BANK_POINTER, AUTO_INCR | LAN91C111_STATUS_BYTES);
+	packet_len = (uint16_t) skb->len;
+	packet_len += LAN91C111_STATUS_BYTES +
+		sizeof(packet_len) + LAN91C111_CTL_BYTES;
+	packet_len &= PLEN_MASK;
+	REG16_STORE(BANK_DATA, packet_len);
+
+	data = (uint8_t *)skb_data_cast_in(skb->data);
+	lan91c111_push_data(data, skb->len);
+
+	if (skb->len % 2) {
+		REG8_STORE(BANK_DATA, ODD_CONTROL);
+	} else {
+		REG16_STORE(BANK_DATA, 0x0);
+	}
+
+	lan91c111_set_cmd(CMD_TX_ENQUEUE);
+
+	return 0;
+}
+
+static int lan91c111_add_pending(struct smc_local *smc, struct sk_buff *skb) {
+	uint8_t int_mask;
+
+	lan91c111_set_bank(2);
+
+	skb_queue_push(&smc->pending_queue, skb);
+
+	int_mask = REG8_LOAD(BANK_INTERRUPT_MASK);
+	int_mask &= ~ALLOC_INT;
```

Comment:
Should it be `int_mask |= ALLOC_INT` instead to enable irq?

Reply:
ooh, thanks a lot.
Done!